stdlib/Test/src/GenericDimensionfuls.jl

```diff
@@ -0,0 +1,109 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module GenericDimensionfuls
+
+export GenericDimensionful
+
+# Here we implement a minimal dimensionful type GenericDimensionful, which is used
+# to test dimensional correctness of various functions in Base.
+
+"""
+The `GenericDimensionful` type can be used to test that numerical code works with
+dimensionful quantities.   Construct a dimensionful quantity with `GenericDimensionful(x)`
+(or `GenericDimensionful{p}(x)` for unitsᵖ).  Arithmetic operations for `GenericDimensionful`
+are defined so that you can only combine quantities in ways that respect the units.
+"""
+struct GenericDimensionful{p,T<:Number} <: Number
+    val::T
+    GenericDimensionful{p,T}(v::Number) where {p,T} = new(v)
+end
+GenericDimensionful(x::T) where {T<:Number} = GenericDimensionful{1,T}(x)
+GenericDimensionful(x::GenericDimensionful) = x
+(::Type{T})(x::GenericDimensionful) where {T<:Number} = T(x.val)::T
+GenericDimensionful{p}(v::Number) where {p} = GenericDimensionful{p,typeof(v)}(v)
+GenericDimensionful{p}(x::GenericDimensionful{q}) where {p,q} = GenericDimensionful{p,typeof(x.val)}(x.val)
+GenericDimensionful{p,T}(x::GenericDimensionful{q}) where {T,p,q} = GenericDimensionful{p,T}(T(x.val))
+
+Base.convert(::Type{GenericDimensionful{p}}, x::GenericDimensionful{p}) where {p} = x
+Base.convert(::Type{GenericDimensionful{p,T}}, x::GenericDimensionful{p}) where {p,T} = GenericDimensionful{p,T}(x)
+Base.convert(::Type{GenericDimensionful{0}}, x::Union{Real,Complex}) = GenericDimensionful{0}(x)
+Base.convert(::Type{GenericDimensionful{0,T}}, x::Union{Real,Complex}) where {T} = GenericDimensionful(convert(T, x))
+Base.convert(D::Type{GenericDimensionful{p}}, x::Number) where {p} = error("dimension mismatch between $D and $(typeof(x))")
+Base.convert(D::Type{GenericDimensionful{p,T}}, x::Number) where {p,T} = error("dimension mismatch between $D and $(typeof(x))")
+
+Base.promote_type(::Type{GenericDimensionful{p,T}}, ::Type{GenericDimensionful{p,S}}) where {p,T,S} =
+    (Base.@_pure_meta; GenericDimensionful{p,promote_type(T,S)})
+
+Base.one(x::GenericDimensionful{p,T}) where {p,T} = one(T)
+Base.one(::Type{GenericDimensionful{p,T}}) where {p,T} = one(T)
+Base.oneunit(x::GenericDimensionful{p,T}) where {p,T} = GenericDimensionful{p,T}(one(T))
+Base.oneunit(x::Type{GenericDimensionful{p,T}}) where {p,T} = GenericDimensionful{p,T}(one(T))
```

Comment:
```suggestion
Base.one(::Type{GenericDimensionful{p,T}}) where {p,T} = one(T)
```
I think this method should suffice.